### ISSUE TITLE
FR: Add Alternative Checkboxes toggle in Style Settings

### ISSUE THREAD

**Issue by damiankorcz** [isaacfreeman/kakano-obsidian-theme#13]
This request comes as part of the initiative I've started for [Obsidian October 2024](https://obsidian.md/blog/2024-obsidian-october/). Additional information and progress on this project can be found here: https://github.com/damiankorcz/Alternative-Checkboxes-Reference-Set

The current goal is to make it easier for users to implement their own CSS Snippets with Alternative Checkboxes, regardless of the theme they use. At the moment, any theme that doesn't present an option to disable their Alternative Checkboxes implementation, requires the snippet to overwrite the existing theme's code. This can become complicated and hard to maintain as currently **44** themes implement Alternative Checkboxes. Ideally, if all themes have the option to disable their implementation, the snippets can be simpler, and work universally across all themes.

If this is something you are interested in implementing, it would require the following code to be added to your theme:

```css
body.enable-alternative-checkboxes {

/* Encapsulate all code relevant to Altenative Checkboxes here */

}
```

Add a toggle in the Style Settings code:

```yaml
-
    id: enable-alternative-checkboxes
    title: Enable Alternative Checkboxes
    description: Disable this if you are using your own implementation via a CSS Snippet.
    default: true
    type: class-toggle
-
```

Since your theme already implements the feature, the code above enables it by default, in order to not change the existing functionality for users.

When the theme implementation is disabled, all additional checkboxes should be styled the same as the default `- [x]` checked checkbox; removing any changes to color, icons, fonts, backgrounds, etc.

If you need any help implementing this, let me know! I can also look into making a Pull Request if you would prefer.

**Comment by isaacfreeman** [comment]
Thanks @damiankorcz, I'll implement this soon and I really like the initiative! I've made an effort with Kakano to provide what I think are the most widely-used and useful alternative checkboxes, and I'd be pleased to support a reference set once that's available.

I plan to put this under a `Compatibility` heading in Style Settings, unless you have any thoughts on how you'd like theme developers to present it. I can also include a brief description linking back to the project page.

The [Theme Dev Vault](https://github.com/obsidian-community/theme-dev-vault) repository would be a good place to document the reference set. I use that vault as a reference during theme development, and I'm sure other theme developers do likewise.

I also like to credit people who have helped me with theme development. Are you happy to be mentioned in the README? I can link to your Github profile, or some other page if you prefer.

**Comment by damiankorcz** [comment]
Hi @isaacfreeman,

At the moment, I'm focusing on getting all themes with Alternative Checkboxes onto common ground so that it's easy to overwrite their implementation before continuing on the Reference Set.

There isn't really a preference as to where the toggle should go. Feel free to put it where you think is the most logical. I'm documenting each one here regardless, so you can have a look where other themes are putting it: 
https://github.com/damiankorcz/Alternative-Checkboxes-Reference-Set/blob/main/09%20Style%20Settings%20Toggle%20Review.md

At the moment I made the vault on my account to keep things together and for an easy way to track the progress. I'll probably look into merging whatever info is important into Obsidian Community repositories such as the Theme Dev Vault.

You are welcome to put me in the README. My GitHub account is fine. Don't feel obliged to as I'm just doing this as a fun project on the side.

**Comment by isaacfreeman** [comment]
Implemented in version 4.2.0.
